### PR TITLE
Add support for creating helm packages without needing the helm binary

### PR DESCRIPTION
**Pull request by paulczar**
Signed-off-by: Paul Czarkowski <username.taken@gmail.com>

### CODE REVIEW

File: cr/cmd/package.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+		client := action.NewPackage()
+		// valueOpts := &values.Options{}
+
+		if config.PackagePath == "" {
+			config.PackagePath = "."
+		}
+		client.Destination = config.PackagePath
+		charts := args
+		if len(charts) == 0 {
+			charts[0] = "."
```
Comment (unguiculus) [inline cr/cmd/package.go:54]: If we want to have this as defaut, we should document it.

File: cr/cmd/package.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+		var err error
+
+		config, err := config.LoadConfiguration(cfgFile, cmd, getRequiredPackageArgs())
+		if err != nil {
+			return err
+		}
+		client := action.NewPackage()
+		// valueOpts := &values.Options{}
+
+		if config.PackagePath == "" {
```
Comment (unguiculus) [inline cr/cmd/package.go:48]: You specified `.cr-release-packages` as default for the Cobra flag. So, this check should probably be removed.

File: cr/cmd/package.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	RunE: func(cmd *cobra.Command, args []string) error {
+		var settings map[string]interface{}
+		var err error
+
+		config, err := config.LoadConfiguration(cfgFile, cmd, getRequiredPackageArgs())
+		if err != nil {
+			return err
+		}
+		client := action.NewPackage()
+		// valueOpts := &values.Options{}
```
Comment (unguiculus) [inline cr/cmd/package.go:46]: Remove

**Comment by paulczar** [comment]
@unguiculus I'm not sure I see the benefit? A user wanting to package a previous release based on a commit can check out that commit, do the package, then check out main again. I feel like adding the need to run `git checkout` twice in the code is potentially fragile.

File: pkg/packager/packager.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+	"helm.sh/helm/v3/pkg/action"
+)
+
+// Packager exposes the packager object
+type Packager struct {
+	config *config.Options
+	paths  []string
+}
+
+// NewPackager does a thing
```
Comment (davidkarlsen) [inline pkg/packager/packager.go:33]:
```suggestion
// NewPackager returns a configured Packager
```

**Comment by paulczar** [comment]
oh no @davidkarlsen your suggested change that I just hit "accept suggestion" didn't have a DCO :skull:

**Comment by paulczar** [comment]
Are we good to move forward with this?   Do we need another reviewer to approve ? @scottrigby

**Comment by scottrigby** [comment]
Hey @paulczar i'm going to try to look over this today

File: cr/cmd/package.go [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+var packageCmd = &cobra.Command{
+	Use:   "package [CHART_PATH] [...]",
+	Short: "Package Helm charts",
+	Long: `This command packages a chart into a versioned chart archive file. If a path
+is given, this will look at that path for a chart (which must contain a
+Chart.yaml file) and then package that directory.
+
+
+If you wish to use advanced packaging options such as creating signed
+packages or updating chart dependencies please use "helm package" instead.`,
```
Comment (scottrigby) [inline cr/cmd/package.go:33]: Looks like we could pass any `helm package` options the way you wrote `CreatePackages()`?

Comment (scottrigby) [reply cr/cmd/package.go:33]: Anyway not saying we need to add that in this PR. But wondering if we could make it easy to encourage users to sign their packages. Would love to see that happen more.

Comment (paulczar) [reply cr/cmd/package.go:33]: Perhaps, but I didn't have the time to test a bunch of that stuff, so I thought for the first PR we should just focus on the most basic use case.

Comment (scottrigby) [reply cr/cmd/package.go:33]: SGTM

**Pull request merged**